majifix311/src/main/java/com/example/majifix311/ui/auth/SigninActivity.java

```diff
@@ -0,0 +1,422 @@
+                new ProgressDialog(SigninActivity.this);
+        progressDialog.setMessage(getString(R.string.auth_label_signin));
+        progressDialog.setIndeterminate(true);
+
+        //show signin progress dialog
+        progressDialog.show();
+
+        Auth auth = Auth.getInstance();
+        auth.signin(email, password)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(new DisposableObserver<Party>() {
+                    @Override
+                    public void onNext(Party party) {
+
+                        //dismiss progress dialog
+                        progressDialog.dismiss();
+
+                        //notify signin succeed
+                        showMessage(R.string.auth_success_signin);
+
+                        //TODO fire party over event bus
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+
+                        //dismiss progress dialog
+                        progressDialog.dismiss();
+
+                        //notify invalid credentials error
+                        if (e instanceof HttpException) {
+                            int code = ((HttpException) e).code();
+                            if (code == 403) {
+                                showMessage(R.string.auth_error_invalid_credentials);
+                            }
+                        }
+
+                        //notify no connection
+                        if (e.getCause() != null && e.getCause() instanceof UnknownHostException) {
+                            showMessage(R.string.auth_error_not_connected);
+                        }
+
+                        //TODO notify other errors
+
+                    }
+
+                    @Override
+                    public void onComplete() {
+                        //finish signin activity
+                        setResult(RESULT_OK);
+                        finish();
+
+                        //TODO add finish smooth animation
+                    }
+
+                });
+    }
+
+
+    /**
+     * Hide/Show email input validation errors as per case
+     *
+     * @param hideOrShow: 1 -> for hide , 2 -> for show
+     */
+    private void hideOrShowEmailValidationError(int hideOrShow) {
+        switch (hideOrShow) {
+            case 1: // for hide error
+                if (emailInputLayout.getChildCount() == 2) {
+                    emailInputLayout.getChildAt(1).setVisibility(View.GONE);
+                }
+                emailInputLayout.setError(null);
+                break;
+            case 2: // for show error
+                if (emailInputLayout.getChildCount() == 2) {
+                    emailInputLayout.getChildAt(1).setVisibility(View.VISIBLE);
+                }
+                emailInputLayout.setError(getString(R.string.auth_error_invalid_email));
+                break;
```

Comment:
What is the need of changing the visibility of the Error to `GONE`, especially as this causes the fields to jump their location. Is this the intended behavior?